cmd/testrunner/sensei.go

```diff
@@ -0,0 +1,243 @@
+	}
+
+	logger.Infow("Attempting to connect to sensei")
+
+	var adminConn *grpc.ClientConn
+	err := tryFunc(
+		func() error {
+			var err error
+			adminConn, err = grpc.Dial(rpcHost, opts...)
+			if err != nil {
+				return err
+			}
+			return err
+		}, 60)
+	if err != nil {
+		return nil, err
+	}
+	defer adminConn.Close()
+
+	logger.Infow("Creating node")
+
+	adminClient := sensei.NewAdminClient(adminConn)
+
+	createResp, err := adminClient.CreateAdmin(context.Background(), &sensei.CreateAdminRequest{
+		Username:   "admin",
+		Alias:      alias,
+		Passphrase: "secret",
+		Start:      true,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	pubkey, err := route.NewVertexFromStr(createResp.Pubkey)
+	if err != nil {
+		return nil, err
+	}
+
+	time.Sleep(time.Second)
+
+	mac := createResp.Macaroon
+
+	cred, err := NewMacaroonCredential(mac)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts,
+		grpc.WithPerRPCCredentials(cred),
+	)
+
+	logger.Infow("Connecting with macaroon")
+
+	var conn *grpc.ClientConn
+	err = tryFunc(
+		func() error {
+			var err error
+			conn, err = grpc.Dial(rpcHost, opts...)
+			if err != nil {
+				return err
+			}
+			return err
+		}, 60)
+	if err != nil {
+		return nil, err
+	}
+
+	return &senseiConnection{
+		alias:  alias,
+		conn:   conn,
+		pubKey: pubkey,
+		client: sensei.NewNodeClient(conn),
+	}, nil
+}
+
+func (l *senseiConnection) PubKey() string {
+	return l.pubKey.String()
+}
+
+func (l *senseiConnection) Close() {
+	l.conn.Close()
+}
+
+func (l *senseiConnection) Connect(key, host string) error {
+	_, err := l.client.ConnectPeer(context.Background(), &sensei.ConnectPeerRequest{
+		NodeConnectionString: fmt.Sprintf("%v@%v", key, host),
+	})
+
+	return err
+}
+
+func (l *senseiConnection) NewAddress() (string, error) {
+	resp, err := l.client.GetUnusedAddress(
+		context.Background(),
+		&sensei.GetUnusedAddressRequest{},
+	)
+	if err != nil {
+		return "", err
+	}
+	return resp.Address, nil
+}
+
+func (l *senseiConnection) OpenChannel(peerKey string, amtSat int64,
+	pushAmtSat int64, private bool) error {
+
+	_, err := l.client.OpenChannels(context.Background(), &sensei.OpenChannelsRequest{
+		Channels: []*sensei.OpenChannelInfo{
+			{
+				AmtSatoshis: uint64(amtSat),
+				Public:      !private,
+
+				// TODO:
+				// NodePubkeyString:   peerKey,
+				// SpendUnconfirmed:   true,
+				// PushSat:            pushAmtSat,
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *senseiConnection) SetPolicy(chanPoint *lnrpc.ChannelPoint,
+	policy *graphreader.PolicyData) error {
+
+	// Not necessary for sender node.
+	panic("not implemented")
+}
+
+func (l *senseiConnection) ActiveChannels() (int, error) {
+	resp, err := l.client.ListChannels(context.Background(), &sensei.ListChannelsRequest{})
+	if err != nil {
+		return 0, err
+	}
+
+	// TODO: How to get numer of active channels?
+	return len(resp.Channels), nil
```

Comment:
You can list channels with a large enough page size (to get them all) and then filter on `is_usable` which means it has enough confirmations and the peer is online.   Can filter on `is_channel_ready` if you only care about channels without enough confirmations.